server/node/drop_table.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package node
+
+import (
+	"fmt"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/plan"
+	"github.com/dolthub/go-mysql-server/sql/rowexec"
+
+	"github.com/dolthub/doltgresql/core"
+	"github.com/dolthub/doltgresql/core/id"
+)
+
+// DropTable is a node that implements functionality specifically relevant to Doltgres' table dropping needs.
+type DropTable struct {
+	gmsDropTable *plan.DropTable
+}
+
+var _ sql.ExecSourceRel = (*DropTable)(nil)
+
+// NewDropTable returns a new *DropTable.
+func NewDropTable(dropTable *plan.DropTable) *DropTable {
+	return &DropTable{
+		gmsDropTable: dropTable,
+	}
+}
+
+// Children implements the interface sql.ExecSourceRel.
+func (c *DropTable) Children() []sql.Node {
+	return c.gmsDropTable.Children()
+}
+
+// IsReadOnly implements the interface sql.ExecSourceRel.
+func (c *DropTable) IsReadOnly() bool {
+	return false
+}
+
+// Resolved implements the interface sql.ExecSourceRel.
+func (c *DropTable) Resolved() bool {
+	return c.gmsDropTable != nil && c.gmsDropTable.Resolved()
+}
+
+// RowIter implements the interface sql.ExecSourceRel.
+func (c *DropTable) RowIter(ctx *sql.Context, r sql.Row) (sql.RowIter, error) {
```

Comment:
Like I said above, I think this is fine for the time being, but we really don't want to make wrapping nodes our interface. Doltgres needs its own Database implementation that handles this stuff.